### PR TITLE
PostgreSQL, MySQL, MariaDB Swagger for the data encryption key api's …

### PR DESCRIPTION
**Pull request by shinilm**
…for BYOK feature

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Review by raych1 — COMMENTED** [review]
@shinilm , next time please add the previous version of swagger as initial commit then update on top of it so that the review will be easier for the diff part. 
Please fix the avocado check failures.

**Comment by shinilm** [comment]
> @shinilm , next time please add the previous version of swagger as initial commit then update on top of it so that the review will be easier for the diff part.
> Please fix the avocado check failures.

This is altogether a new api under and new api version and hence this was newly added.

File: specification/mariadb/resource-manager/Microsoft.DBforMariaDB/preview/2020-01-01-privatepreview/DataEncryptionKeys.json [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+          "enum": [
+            "AzureKeyVault"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "ServerKeyType",
+            "modelAsString": true
+          }
+        },
+        "uri": {
```
Comment (ryansbenson) [inline specification/mariadb/resource-manager/Microsoft.DBforMariaDB/preview/2020-01-01-privatepreview/DataEncryptionKeys.json]: What is this uri? Can it be anything or only a keyvault key?

How do you access keyvault? Do you plan to use MSI to access this key?

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2020-01-01-privatepreview/DataEncryptionKeys.json [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+          "enum": [
+            "AzureKeyVault"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "ServerKeyType",
+            "modelAsString": true
+          }
+        },
+        "uri": {
```
Comment (ryansbenson) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2020-01-01-privatepreview/DataEncryptionKeys.json]: Same URI comment.

File: specification/mariadb/resource-manager/Microsoft.DBforMariaDB/preview/2020-01-01-privatepreview/DataEncryptionKeys.json (hunk 1 above)
Comment (ryansbenson) [reply specification/mariadb/resource-manager/Microsoft.DBforMariaDB/preview/2020-01-01-privatepreview/DataEncryptionKeys.json]: looking at your RP manifest change, you'll expose MSI on the /server resource. You should document this in swagger too.

Comment (shinilm) [reply specification/mariadb/resource-manager/Microsoft.DBforMariaDB/preview/2020-01-01-privatepreview/DataEncryptionKeys.json]: This Uri is Azure key vault.  Yes.  We are using MSI to access this key

Comment (shinilm) [reply specification/mariadb/resource-manager/Microsoft.DBforMariaDB/preview/2020-01-01-privatepreview/DataEncryptionKeys.json]: > looking at your RP manifest change, you'll expose MSI on the /server resource. You should document this in swagger too.

Done

Comment (shinilm) [reply specification/mariadb/resource-manager/Microsoft.DBforMariaDB/preview/2020-01-01-privatepreview/DataEncryptionKeys.json]: This Uri is Azure key vault. Yes. We are using MSI to access this key. Included the resource identity as well in the manifest

File: specification/mariadb/resource-manager/Microsoft.DBforMariaDB/preview/2020-01-01-privatepreview/DataEncryptionKeys.json [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+    },
+    "ServerKeyProperties": {
+      "description": "Properties for a key execution.",
+      "required": [
+        "serverKeyType"
+      ],
+      "type": "object",
+      "properties": {
+        "serverKeyType": {
+          "description": "The key type like 'ServiceManaged', 'AzureKeyVault'.",
```
[the remark this answers is not in the document]
Comment (shinilm) [reply specification/mariadb/resource-manager/Microsoft.DBforMariaDB/preview/2020-01-01-privatepreview/DataEncryptionKeys.json]: Right now we are only accepting Azure key vault uri

File: specification/mariadb/resource-manager/Microsoft.DBforMariaDB/preview/2020-01-01-privatepreview/DataEncryptionKeys.json (hunk 1 above)
Comment (ryansbenson) [reply specification/mariadb/resource-manager/Microsoft.DBforMariaDB/preview/2020-01-01-privatepreview/DataEncryptionKeys.json]: You should call this property (and the other 'uri' one) something more specific. Are you planning to add a LinkedAccessCheck to this? If so I'd make it specific to the purpose of keyvault resource id. When you introduce more serverKeyTypes, you can use discriminators to change the schema to be specific to each type introduced.

Comment (shinilm) [reply specification/mariadb/resource-manager/Microsoft.DBforMariaDB/preview/2020-01-01-privatepreview/DataEncryptionKeys.json]: As discussed over the call, we will address these before public preview

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/stable/2017-12-01/mysql.json [hunk 4]
```diff
@@ -1615,8 +1615,42 @@
...
+          "format": "uuid",
+          "description": "The Azure Active Directory tenant id.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
     "Server": {
       "properties": {
+        "identity": {
```
[the remark this answers is not in the document]
Comment (shinilm) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/stable/2017-12-01/mysql.json]: Resolved this as per the call

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2017-12-01/postgresql.json [hunk 5]
```diff
@@ -1619,8 +1619,42 @@
...
+          "format": "uuid",
+          "description": "The Azure Active Directory tenant id.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
     "Server": {
       "properties": {
+        "identity": {
```
[the remark this answers is not in the document]
Comment (shinilm) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2017-12-01/postgresql.json]: Resolved this as per the call

File: specification/mariadb/resource-manager/Microsoft.DBforMariaDB/stable/2018-06-01/mariadb.json [hunk 6]
```diff
@@ -1619,8 +1619,42 @@
...
+          "format": "uuid",
+          "description": "The Azure Active Directory tenant id.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
     "Server": {
       "properties": {
+        "identity": {
```
[the remark this answers is not in the document]
Comment (shinilm) [reply specification/mariadb/resource-manager/Microsoft.DBforMariaDB/stable/2018-06-01/mariadb.json]: Resolved this as per the call

**Review by ryansbenson — APPROVED** [review]
Approved from ARMs side

**Review by raych1 — APPROVED** [review]

**Pull request merged**